frappe/model/rename_doc.py

```diff
@@ -90,18 +91,27 @@ def update_document_title(
 			doc.rename(updated_name, merge=merge)
 
 	if title_updated:
-		try:
-			setattr(doc, title_field, updated_title)
-			doc.save()
-			frappe.msgprint(_("Saved"), alert=True, indicator="green")
-		except Exception as e:
-			if frappe.db.is_duplicate_entry(e):
-				frappe.throw(
-					_("{0} {1} already exists").format(doctype, frappe.bold(docname)),
-					title=_("Duplicate Name"),
-					exc=frappe.DuplicateEntryError,
-				)
-			raise
+		if action_enqueued:
+			frappe.enqueue(
+				"frappe.client.set_value",
+				doctype=doc.doctype,
+				name=updated_name,
+				fieldname=title_field,
```

Comment:
The title field never needs enqueueing AFAIk. It's just ~1 update query in DB.

Reply:
The rationale is to stall the title update action until the document is renamed.

Reply:
In that case just changing title shouldn't go to background (does rn)